### PR TITLE
data and scripts for amazon-google dataset 

### PR DESCRIPTION
**Pull request by navinrathore**

### CODE REVIEW

**Comment by navinrathore** [comment]
Run the pyspark scripts from dirctory - examples/amazon-google/scripts
 spark-submit prepareTestDataSpark.py
 
 training file has cluster_id and label at the end right now. trying to move them in the beginning...selection column list needs to be adjusted accordingly.

File: examples/amazon-google/config.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
+{
```
Comment (sonalgoyal) [inline examples/amazon-google/config.json:1]: why do we need two configs - dont we need just one for linking?

File: examples/amazon-google/configWithTrainingSamples.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+					{\"name\":\"title\", \"type\":\"string\", \"nullable\":true},
+					{\"name\":\"manufacturer\",\"type\":\"string\",\"nullable\":true} ,
+					{\"name\":\"price\", \"type\":\"double\", \"nullable\":true}
+				]
+			}"
+		}
+		],
+		"labelDataSampleSize" : 0.4,
+		"numPartitions":4,
+		"modelId": 1101,
```
Comment (sonalgoyal) [inline examples/amazon-google/configWithTrainingSamples.json:99]: our model series is 100

File: examples/amazon-google/configWithTrainingSamples.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		"zinggDir": "models",
+		"collectMetrics": false
```
Comment (sonalgoyal) [inline examples/amazon-google/configWithTrainingSamples.json:101]: this flag should be removed.

File: examples/amazon-google/scripts/prepareTestDataSpark.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+# def getClusterId(x) :
+#     #return '{0: >8}'.format(id1) + ":" + '{0: >8}'.format(id2)
+#     return x['ltable_id'] + ":" + x['rtable_id']
+
+def getClusterId(x) :
+	#return '{0: >8}'.format(id1) + ":" + '{0: >8}'.format(id2)
+	cluster_id = x.ltable_id + ":" + x.rtable_id
+	return (x.ltable_id, x.rtable_id, x.label, cluster_id)
+
+tableA="../dataset_orig/tableA.csv"
```
Comment (sonalgoyal) [inline examples/amazon-google/scripts/prepareTestDataSpark.py:36]: rename script location folder to original

File: examples/amazon-google/scripts/prepareTrainDataSpark.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+tableA="../dataset_orig/tableA.csv"
+tableB="../dataset_orig/tableB.csv"
+train="../dataset_orig/train.csv"
+# test="../dataset_orig/test.csv"
+valid="../dataset_orig/valid.csv"
+
+train_out = "../train.csv"
+
+def readCSVFile(spark, fileName) :
```
Comment (sonalgoyal) [inline examples/amazon-google/scripts/prepareTrainDataSpark.py:39]: why not have all the methods in one script and use them? Ideally these scripts should work for other datasets too.

**Review by sonalgoyal — CHANGES_REQUESTED** [review]

File: examples/amazon-google/config.json (hunk 1 above)
Comment (navinrathore) [reply examples/amazon-google/config.json]: Removed.
It was added if somebody wants to generate own training/labelled data.

File: examples/amazon-google/scripts/prepareTestDataSpark.py (hunk 4 above)
Comment (navinrathore) [reply examples/amazon-google/scripts/prepareTestDataSpark.py]: removed

File: examples/amazon-google/scripts/prepareTrainDataSpark.py (hunk 5 above)
Comment (navinrathore) [reply examples/amazon-google/scripts/prepareTrainDataSpark.py:34]: It will work for other datasets also.
They will be merged.

**Review by sonalgoyal — COMMENTED** [review]
what is the need for config.json?

**Comment by navinrathore** [comment]
The code was updated yesterday with some changes. Some improvements were made and some mistakes were rectified.
Some analysis of the results was also done.
**Result**
Test1 (Google): Total 2293 records -> 771 unique  ==> After Linking 565 records found matched
Test2 (AWS)   : Total 2293 records -> 1090 unique  ==> 2287 records found matched
Observations:  in teset2 records (we did have matched records, we had other partially matched records). Perhaps updating model with some parameters, results can be improved. the score/probability was around 0.45 mostly.
Training data: both train,csv (6874 recs)and valid.csv(2293) was used to train the model.

File: examples/amazon-google/scripts/prepareTestDataSpark.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+tableB="../dataset_orig/tableB.csv"
+train="../dataset_orig/train.csv"
+test="../dataset_orig/test.csv"
+valid="../dataset_orig/valid.csv"
+
+testA_out = "../testA.csv"
+testB_out = "../testB.csv"
+
+def init() :
```
[the remark this answers is not in the document]
Comment (navinrathore) [reply examples/amazon-google/scripts/prepareTestDataSpark.py]: It's outdated. already cleaned.

File: examples/amazon-google/config.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+				\"fields\" : [ 
+					{\"name\":\"id\", \"type\":\"string\", \"nullable\":false}, 
+					{\"name\":\"title\", \"type\":\"string\", \"nullable\":true},
+					{\"name\":\"manufacturer\",\"type\":\"string\",\"nullable\":true},
+					{\"name\":\"price\", \"type\":\"double\", \"nullable\":true}
+					]
+					}"
+				},
+				{
+					"name": "test2",
```
[the remark this answers is not in the document]
Comment (navinrathore) [reply examples/amazon-google/config.json]: ok

File: examples/amazon-google/config.json [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+			"props": {
+				"location": "/tmp/zinggOutput",
+				"delimiter": ",",
+				"header": false
+			}
+		}
+	],
+	"data": [
+		{
+			"name": "test1",
```
[the remark this answers is not in the document]
Comment (navinrathore) [reply examples/amazon-google/config.json]: ok

File: examples/amazon-google/configWithTrainingSamples.json (hunk 3 above)
Comment (navinrathore) [reply examples/amazon-google/configWithTrainingSamples.json]: already done.

File: examples/amazon-google/config.json (hunk 1 above)
Comment (navinrathore) [reply examples/amazon-google/config.json]: File removal will be part of next checkin. Version from first checkin is still there.

**Comment by navinrathore** [comment]
### Some queries/notes about changes:
* field 'price' is 'exact'/double and all others are 'fuzzy'/string
* In configWithTrainingSample.json, the training sample data comprises both train.csv and valid.csv data
* **In configWithTrainingSample.json, is "data" attributes compulsory? it currently has both tableA and tableB. Or it could have data corresponding to train.csv and valid.csv**
*  **What could be ideal location of the script? Some minor changes can be made it absolutely generic in terms of location**
*  Same script works with all dataset without any change. 
* Naming is simple testA, TestB etc.

File: examples/amazon-google/scripts/prepareTrainDataSpark.py (hunk 5 above)
Comment (navinrathore) [reply examples/amazon-google/scripts/prepareTrainDataSpark.py]: There is only one script now

File: examples/amazon-google/config.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (navinrathore) [reply examples/amazon-google/config.json]: Yes. I've made the changes. 
For other datasets, these fields may be updated accordingly.

**Pull request merged**